checks/evaluation/branch_protection.go

```diff
@@ -362,6 +326,18 @@ func adminReviewProtection(branch *clients.BranchRef, dl checker.DetailLogger) (
 		}
 	}
 
+	if isUserAdmin(branch) {
```

Comment:
I acknowledge that this implementation is far from ideal, it's basically based on my interpretation on GitHub's responses and we probably don't have guarantees that they'll keep this same pattern. That said, I'll very open for better ideas on how to implement this.

I coded this way basically because I couldn't find a better way to get this data from the information available, and with the movement to the usage of GitHub Repo Rules, we shouldn't need this distinction between admin/non-admin for long.